lib/api.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Specberus REST API.
+ */
+
+// Internal packages:
+const package = require('../package.json')
+,   sink = require('./sink')
+,   util = require('./util')
+,   validator = require('./validator')
+;
+
+const Sink = sink.Sink
+,   profiles = util.profiles
+,   version = package.version
+;
+
+/**
+ * Build an "options" object based on an HTTP query string.
+ *
+ * @param {Object} query - an HTTP request query.
+ * @returns {Object} an "options" object that can be used by Specberus.
+ */
+
+const parseSource = function(query) {
+    const result = {};
+    if (query.url) result.url = query.url;
+    else if (query.source) result.source =uery.source;
+    else if (query.file) result.file = query.file;
+    else if (query.document) result.document = query.document;
+    if (query.profile) {
+        if (profiles[query.profile])
+            result.profile = profiles[query.profile];
+        else
+            throw new Error('Cannot retrieve profile “' + query.profile + '”');
+    }
+    if (query.validation) result.validation = query.validation;
+    if (query.processDocument) result.processDocument = query.processDocument;
```

Comment:
You are right, I should be accepting the complete set of parameters.

@deniak, I'm now refactoring the parsing of parameters, because otherwise it'd be duplicated in different places. That common logic will be used then by `lib/validator.js` and `lib/api.js` to parse their parameters. Sounds good?